tests/pos/i11081.scala

```diff
@@ -0,0 +1,5 @@
+enum Outer:
+  case Foo
+object Outer:
+  trait Bar
+  case class Baz(bar: Bar)
```

Comment:
it still fails if instead there is a default argument to Baz's constructor that references Bar, I can add the extra check

Reply:
Ah yes, I overlooked that part. Well spotted.